protocol/genesis/types/share_test.go

```diff
@@ -0,0 +1,107 @@
+package types
+
+import (
+	"bytes"
+	"testing"
+
+	spectypes "github.com/ssvlabs/ssv-spec/types"
+	"github.com/stretchr/testify/require"
+)
+
+func TestConvertShare(t *testing.T) {
+	// Step 1: Create an Alan share with all fields populated
+	originalShare := &spectypes.Share{
+		ValidatorPubKey: [48]byte{1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14, 15, 16, 17, 18, 19, 20, 21, 22, 23, 24, 25, 26, 27, 28, 29, 30, 31, 32, 33, 34, 35, 36, 37, 38, 39, 40, 41, 42, 43, 44, 45, 46, 47, 48},
+		SharePubKey:     []byte{49, 50, 51, 52, 53, 54, 55, 56, 57, 58, 59, 60},
+		Committee: []*spectypes.ShareMember{
+			{SharePubKey: []byte{61, 62, 63, 64, 65, 66, 67, 68, 69, 70, 71, 72}, Signer: 1},
+			{SharePubKey: []byte{73, 74, 75, 76, 77, 78, 79, 80, 81, 82, 83, 84}, Signer: 2},
+			{SharePubKey: []byte{85, 86, 87, 88, 89, 90, 91, 92, 93, 94, 95, 96}, Signer: 3},
+			{SharePubKey: []byte{97, 98, 99, 100, 101, 102, 103, 104, 105, 106, 107, 108}, Signer: 4},
+		},
+		DomainType:          [4]byte{109, 110, 111, 112},
+		FeeRecipientAddress: [20]byte{113, 114, 115, 116, 117, 118, 119, 120, 121, 122, 123, 124, 125, 126, 127, 128, 129, 130, 131, 132},
+		Graffiti:            []byte{133, 134, 135, 136, 137, 138, 139, 140, 141, 142, 143, 144},
+	}
+
+	// Step 2: Convert to Genesis share
+	genesisShare := ConvertToGenesisShare(originalShare, &spectypes.CommitteeMember{OperatorID: 1})
+
+	// Step 3: Convert share back to Alan share
+	convertedBackShare := ConvertFromGenesisShare(genesisShare)
+
+	// Step 4: Expect equality with original share
+	require.True(t, sharesEqual(originalShare, convertedBackShare), "The converted back share should be equal to the original share")
+
+	// Step 5: Modify final share and expect inequality
+	modifiedShare := deepCopyShare(convertedBackShare)
```

Comment:
the fact that we need to copy the converted share before modifying to not modify the original has raised a bug in the conversion functions — they don't properly copy all of the fields.

instead of copying, let's fix the conversion funcs to copy any non-value field (such as slices or pointers) before assigning it to the new share.

Reply:
done here - https://github.com/ssvlabs/ssv/pull/1563/commits/b58bda9f5596a412202a2057ed3dcf03c9796b54